java/ql/src/experimental/Security/CWE/CWE-600/UncaughtServletException.ql

```diff
@@ -9,36 +9,60 @@
 
 import java
 import semmle.code.java.dataflow.FlowSources
+import semmle.code.java.dataflow.DataFlow2
 import semmle.code.java.dataflow.TaintTracking
 import semmle.code.java.frameworks.Servlets
 import semmle.code.xml.WebXML
 import DataFlow::PathGraph
 
+/** DataFlow configuration for detecting the case of rethrowing unprocessed exceptions */
+class ThrowExConfiguration extends DataFlow2::Configuration {
+  ThrowExConfiguration() { this = "Throwing Unprocessed Exception Configuration" }
+
+  /** Source of `LocalVariableDeclExpr` */
+  override predicate isSource(DataFlow::Node source) {
+    exists(LocalVariableDeclExpr v | source.asExpr() = v.getAnAccess())
```

Comment:
This might not be needed because the config is only used when the `source` is known to be a `LocalVariableDeclExpr` access.

Reply:
@Marcono1234 Thanks for the prompt update. CodeQL requires <code>override predicate isSource</code> for a data flow configuration. Could you provide more details on what's the desired change?

Also please excuse my ignorance - I don't know where to dismiss your review comments. I thought "resolve" will do. Could you also provide more details as well?

Reply:
In line 65 where you are using `hasFlow` the source is already constrained to be an access of a catch variable, therefore the check here in `isSource` might be redundant. Since the predicate has to be overridden, you could use `any()` (with a comment explaining the reason) I guess.
Though it likely does not matter much and your current solution might be better understandable, so it is likely fine to leave it as is unless the maintainers want this to be changed.

----

Sorry for the confusion regarding my review comments. What I meant is that I am not a maintainer / member of this project. So if my review comments are bothering you or you are getting frustrated due to them I do not mind if you just ignore / resolve them without performing any changes.

Reply:
In another query, I used <code>any()</code> but was advised to make it specific, which is better understandable. So let's leave it for now.

And thanks for all your help with reviewing this PR. Teamwork makes it useful and shiny!